src/main/java/Backend/FileDownloader.java

```diff
@@ -386,31 +408,27 @@ public static void downloadFromInstagram(String link, String dir) throws Interru
         } else {
             fileDownloadMessagePart = outputFileName;
         }
-
-        ProcessBuilder processBuilder;
-
-        /*
-         * Create a new ProcessBuilder object.
-         *
-         * The command line arguments tell `yt-dlp` to download the video or photo in MP4 format and to save it to the specified directory.
-         */
+        ProcessBuilder processBuilder; // Creates a new ProcessBuilder object
         messageBroker.sendMessage("Trying to download " + fileDownloadMessagePart + " ...", LOGGER_INFO, "download");
         Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize(); // E.g. java.awt.Dimension[width=1366,height=768]
         int height = (int) screenSize.getHeight();  // E.g.: 768
         int width = (int) screenSize.getWidth();    // E.g.: 1366
-        String query = link.substring(link.indexOf("https://instagram.com/p/") + 17);
-        query = query.substring(0, query.indexOf("/"));
         if (dir.length() == 0 || (dir.equalsIgnoreCase("."))) {
-            processBuilder = new ProcessBuilder("yt-dlp", "--quiet", "--progress", "https://instagram.com/p/" + query + "/mp4", "-o",
-                    outputFileName, "-f", "[height<=" + height + "][width<=" + width + "]");
+            processBuilder = new ProcessBuilder(dirOfYt_dlp + yt_dlpProgramName, "--quiet", "--progress", link, "-o",
```

Comment:
Okay, so if I'm right then here you're directly passing the link variable in the 'yt-dlp' command instead of constructing the URL by extracting the necessary information from the link (the way I approached).